source/elements/oneDAL/include/oneapi/dal/common.hpp

```diff
@@ -0,0 +1,58 @@
+namespace oneapi::dal {
+
+class base {
+public:
+    virtual ~base() = default;
+};
+
+enum class data_type {
+    /// 8-bit signed integer value type
+    int8,
+    /// 16-bit signed integer value type
+    int16,
+    /// 32-bit signed integer value type
+    int32,
+    /// 64-bit signed integer value type
+    int64,
+    /// 8-bit unsigned integer value type
+    uint8,
+    /// 16-bit unsigned integer value type
+    uint16,
+    /// 32-bit unsigned integer value type
+    uint32,
+    /// 64-bit unsigned integer value type
+    uint64,
+    /// 32-bit floating-point value type
+    float32,
+    /// 64-bit floating-point value type
+    float64,
+    /// bi-float value type
+    bfloat16
+};
+
+struct range {
```

Comment:
it potentially would be used in graph as well

Reply:
Now they use their own range